app/src/main/kotlin/com/intive/picover/common/di/AppModule.kt

```diff
@@ -10,9 +12,13 @@ import kotlinx.coroutines.Dispatchers
 @Module
 @InstallIn(SingletonComponent::class)
 object AppModule {
+
 	@Provides
 	fun provideFirebaseStoreInstance() = FirebaseStorage.getInstance()
 
+	@Provides
+	fun provideFirebaseDatabase() = Firebase.database
```

Comment:
Yes, i agree.

But i am wondering, if we should create a separate, firebasse  module, or firebaseDatabase and firebaseStorage modules, as we have separate module for auth. 

I am not sure about the correct path of growing on this level of implementation

Reply:
I think that even without our `SingletonComponent` module the Firebase is a singleton, and every call to the `Firebase.database` will provide us with the same instance. Fully agree with the fact that we should not put everything in the app module, but in my opinion, in the case of such dependencies like Firebase, it's OK to put them in the app module. Is there any benefit of extracting it to a separate module, or making it for example `ViewModelComponent`? Another thing is that it will be easier in the future to split the `AppModule` into a few other modules than to combine 5 different redundant modules with 1-2 provides into one. If we are not sure how we want to divide such dependencies, it is better to start with the simplest solution.

Reply:
I wouldn't agree, nice example of messing `AppModule`, is our project. For everyone it's easier to put everything in one `SingletonComponent` module, because it's the easiest way to make it work without any errors, but also without thinking what component and scoping should be used.  
Indeed, it will be easier to combine 5 different modules, with different components and scoping, than splitting big `AppModule`, and most likely nobody would like to handle such splitting because it's a painful when such module will grow too much. 
I would not block current approach, it's not a big mistake, but for me it's not appropriate in a long term.

Reply:
So I think it's a nice candidate to encourage all team members to think about that and then we can discuss it. For me at that point is totally okay to go with the simplest solution :)